wait/tls_test.go

```diff
@@ -0,0 +1,146 @@
+package wait_test
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	_ "embed"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/errdefs"
+	"github.com/stretchr/testify/require"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	serverName         = "127.0.0.1"
+	caFilename         = "/tmp/ca.pem"
+	clientCertFilename = "/tmp/cert.crt"
+	clientKeyFilename  = "/tmp/cert.key"
+)
+
+var (
+	//go:embed testdata/cert.crt
```

Comment:
question: Have you considered using https://github.com/mdelapenya/tlscert instead?

Reply:
I saw that, but it was hiding errors e.g. [SelfSignedFromRequest](https://pkg.go.dev/github.com/mdelapenya/tlscert#SelfSignedFromRequest) returns nil on error, so followed the pattern in the standard library. If that got cleaned up we could change later.